sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs

```diff
@@ -36,13 +35,49 @@ public partial class ConfigurationClient
 
         private static async Task<Response<ConfigurationSetting>> CreateResponseAsync(Response response, CancellationToken cancellation)
         {
-            ConfigurationSetting result = await ConfigurationServiceSerializer.DeserializeSettingAsync(response.ContentStream, cancellation).ConfigureAwait(false);
+            ConfigurationSetting result;
+
+            if (response.ContentStream == null || response.ContentStream.Length == 0)
+            {
+                result = CreateSettingFromHeaders(response);
+            }
+            else
+            {
+                result = await ConfigurationServiceSerializer.DeserializeSettingAsync(response.ContentStream, cancellation).ConfigureAwait(false);
+            }
+
             return new Response<ConfigurationSetting>(response, result);
         }
 
         private static Response<ConfigurationSetting> CreateResponse(Response response)
         {
-            return new Response<ConfigurationSetting>(response, ConfigurationServiceSerializer.DeserializeSetting(response.ContentStream));
+            ConfigurationSetting result;
+
+            if (response.ContentStream == null || response.ContentStream.Length == 0)
+            {
+                result = CreateSettingFromHeaders(response);
+            }
+            else
+            {
+                result = ConfigurationServiceSerializer.DeserializeSetting(response.ContentStream);
+            }
+
+            return new Response<ConfigurationSetting>(response, result);
+        }
+
+        private static ConfigurationSetting CreateSettingFromHeaders(Response response)
```

Comment:
Not a fan of this pattern, just get the etag header value in `HasChanged` method itself